src/app/routes/connector-ui/asset-page/asset-edit-dialog/asset-edit-dialog.component.ts

```diff
@@ -0,0 +1,124 @@
+import {Component, Inject, OnDestroy} from '@angular/core';
+import {MAT_DIALOG_DATA, MatDialogRef} from '@angular/material/dialog';
+import {EMPTY, Observable, Subject, switchMap} from 'rxjs';
+import {catchError, finalize, map, takeUntil, tap} from 'rxjs/operators';
+import {IdResponseDto} from '@sovity.de/edc-client';
+import {EdcApiService} from '../../../../core/services/api/edc-api.service';
+import {AssetCreateRequestBuilder} from '../../../../core/services/asset-create-request-builder';
+import {AssetService} from '../../../../core/services/asset.service';
+import {NotificationService} from '../../../../core/services/notification.service';
+import {ValidationMessages} from '../../../../core/validators/validation-messages';
+import {AssetEditDialogData} from './asset-edit-dialog-data';
+import {AssetEditDialogResult} from './asset-edit-dialog-result';
+import {AssetAdvancedFormBuilder} from './form/asset-advanced-form-builder';
+import {AssetDatasourceFormBuilder} from './form/asset-datasource-form-builder';
+import {AssetEditDialogForm} from './form/asset-edit-dialog-form';
+import {AssetMetadataFormBuilder} from './form/asset-metadata-form-builder';
+import {AssetEditorDialogFormValue} from './form/model/asset-editor-dialog-form-model';
+
+@Component({
+  selector: 'asset-edit-dialog',
+  templateUrl: './asset-edit-dialog.component.html',
+  providers: [
+    AssetAdvancedFormBuilder,
+    AssetDatasourceFormBuilder,
+    AssetEditDialogForm,
+    AssetCreateRequestBuilder,
+    AssetMetadataFormBuilder,
+  ],
+})
+export class AssetEditDialogComponent implements OnDestroy {
+  loading = false;
+
+  methods = ['GET', 'POST', 'PUT', 'PATCH', 'DELETE', 'OPTIONS', 'HEAD'];
```

Comment:
Should this be defined as a constant somewhere else? seems like it might be used in other places at some point

Reply:
the list of methods is selectable because there is an "initiate transfer dialog" somewhere else that for example cannot use GET as the GET method cannot have a request body (which would need to contain the data)

note that this class did not get tracked as renamed. I renamed "asset-create-dialog" to "asset-edit-dialog" as it handles both creating and editing